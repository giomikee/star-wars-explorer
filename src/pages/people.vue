<template>
  <v-container>
    <h2>People Database</h2>
  </v-container>

  <SearchNameInput
    :is-loading="arePeopleEmpty"
    @name-searched="setSearchedName"
  />

  <SortButtons
    :is-loading="arePeopleEmpty"
    @sorted:date="sortPeopleByDate"
    @sorted:name="sortPeopleByName"
  />

  <TableWrapper
    :current-page="currentPage"
    :is-loading="arePeopleEmpty"
    :pages="itemsPages"
    @page-changed="updateCurrentPage"
  >
    <PeopleTable :current-page="currentPage" />
  </TableWrapper>
</template>

<script setup lang="ts">
  import PeopleTable from '../components/PeopleTable/PeopleTable.vue'
  import SearchNameInput from '../components/SearchNameInput/SearchNameInput.vue'
  import TableWrapper from '../components/TableWrapper/TableWrapper.vue'
  import { usePeoplePage } from './composables'

  const {
    arePeopleEmpty,
    currentPage,
    itemsPages,
    setSearchedName,
    sortPeopleByDate,
    sortPeopleByName,
    updateCurrentPage,
  } = usePeoplePage()
</script>
