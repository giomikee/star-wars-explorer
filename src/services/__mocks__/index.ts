export * from './people.service'
export * from './planets.service'
