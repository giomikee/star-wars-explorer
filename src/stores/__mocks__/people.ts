import type { People } from '../../interfaces'

export const mockedPeople: People[] = [
  {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person2',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person3',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person4',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person5',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person6',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person7',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person8',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person9',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person10',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person11',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person12',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
]
