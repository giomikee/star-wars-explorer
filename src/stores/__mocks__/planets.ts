import type { Planet } from '../../interfaces'

const mockedDate = new Date('2014-12-09T13:50:49.641000Z')

export const mockedPlanets: Planet[] = [
  {
    name: 'Tatooine',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet2',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet3',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet4',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet5',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet6',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet7',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet8',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet9',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet10',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet11',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet12',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
]
