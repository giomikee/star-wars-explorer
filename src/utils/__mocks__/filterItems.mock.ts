import type { Item } from '../../interfaces'

export const mockedDate = new Date('2014-12-09T13:50:49.641000Z')

export const mockedItems: Item[] = [
  {
    name: 'Foo',
    created: mockedDate,
  },
  {
    name: 'Food',
    created: mockedDate,
  },
  {
    name: 'Bar',
    created: mockedDate,
  },
]
