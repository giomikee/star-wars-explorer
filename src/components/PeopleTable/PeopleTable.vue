<template>
  <v-table class="w-100">
    <thead>
      <tr>
        <th scope="col">Name</th>
        <th scope="col">Height</th>
        <th scope="col">Mass</th>
        <th scope="col">Hair Color</th>
        <th scope="col">Eye Color</th>
        <th scope="col">Birth Year</th>
        <th scope="col">Gender</th>
        <th scope="col">Date Added</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="people in paginatedPeople" :key="people.url">
        <td>{{ people.name }}</td>
        <td>{{ people.height }}</td>
        <td>{{ people.mass }}</td>
        <td>{{ people.hair_color }}</td>
        <td>{{ people.eye_color }}</td>
        <td>{{ people.birth_year }}</td>
        <td>{{ people.gender }}</td>
        <td>{{ formatDateTime(people.created) }}</td>
      </tr>
    </tbody>
  </v-table>
</template>

<script lang="ts" setup>
  import type { People } from '../../interfaces'
  import { computed } from 'vue'
  import { usePeopleStore } from '../../stores/people.store'
  import { formatDateTime, getPaginatedItems } from '../../utils'

  interface Props {
    currentPage: number
  }

  const props = defineProps<Props>()
  const { getPeopleToDisplay } = usePeopleStore()

  const paginatedPeople = computed<People[]>(() => getPaginatedItems(getPeopleToDisplay(), props.currentPage))
</script>

<script lang="ts">
  export default { name: 'PeopleTable' }
</script>
