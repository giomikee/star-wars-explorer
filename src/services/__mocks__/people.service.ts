import { mockedPeople } from '../../stores/__mocks__/people'

export function getAllPeople () {
  return mockedPeople
}
