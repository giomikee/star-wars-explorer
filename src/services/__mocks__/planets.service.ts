import { mockedPlanets } from '../../stores/__mocks__/planets'

export function getAllPlanets () {
  return mockedPlanets
}
