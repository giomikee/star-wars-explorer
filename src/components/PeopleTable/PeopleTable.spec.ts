import type { People } from '../../interfaces'
import { shallowMount } from '@vue/test-utils'
import { beforeAll, describe, expect, it, vi } from 'vitest'
import { mockedPeople } from '../../stores/__mocks__/people'
import { usePeopleStore } from '../../stores/people.store'
import PeopleTable from './PeopleTable.vue'

const defaultProps = { currentPage: 1 }

interface Component {
  paginatedPeople: People[]
}

const createWrapper = (props = defaultProps) => shallowMount(
  PeopleTable,
  {
    props,
    global: {
      stubs: {
        VTable: false,
      },
    },
  })

describe('Given PeopleTable component', () => {
  beforeAll(() => {
    const store = usePeopleStore()
    vi.spyOn(store, 'getPeopleToDisplay').mockReturnValue(mockedPeople)
  })

  describe('When component is rendered', () => {
    it('Then it should match snapshot', () => {
      const { element } = createWrapper()

      expect(element).toMatchSnapshot()
    })
  })

  describe('Given computed properties', () => {
    describe.each([
      {
        currentPage: 1,
        expectedLength: 10,
      },
      {
        currentPage: 2,
        expectedLength: 2,
      },
    ])('When paginatedPeople is evaluated', ({ currentPage, expectedLength }) => {
      describe(`And currentPage prop is ${currentPage}`, () => {
        it(`Then it should return ${expectedLength} people from the store`, () => {
          const wrapper = createWrapper({ currentPage })

          expect((wrapper.vm as typeof wrapper.vm & Component).paginatedPeople).toHaveLength(expectedLength)
        })
      })
    })
  })
})
