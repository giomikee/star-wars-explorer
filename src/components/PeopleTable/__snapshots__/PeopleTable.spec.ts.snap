// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given PeopleTable component > When component is rendered > Then it should match snapshot 1`] = `
<div
  class="v-table v-theme--dark v-table--density-default w-100"
>
  <!---->
  <div
    class="v-table__wrapper"
  >
    <table>
      
      <thead>
        <tr>
          <th
            scope="col"
          >
            Name
          </th>
          <th
            scope="col"
          >
            Height
          </th>
          <th
            scope="col"
          >
            Mass
          </th>
          <th
            scope="col"
          >
            Hair Color
          </th>
          <th
            scope="col"
          >
            Eye Color
          </th>
          <th
            scope="col"
          >
            Birth Year
          </th>
          <th
            scope="col"
          >
            Gender
          </th>
          <th
            scope="col"
          >
            Date Added
          </th>
        </tr>
      </thead>
      <tbody>
        
        <tr>
          <td>
            Luke Skywalker
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person2
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person3
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person4
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person5
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person6
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person7
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person8
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person9
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        <tr>
          <td>
            Person10
          </td>
          <td>
            172
          </td>
          <td>
            77
          </td>
          <td>
            blond
          </td>
          <td>
            blue
          </td>
          <td>
            19BBY
          </td>
          <td>
            male
          </td>
          <td>
            December 20, 2014 at 10:17 PM
          </td>
        </tr>
        
      </tbody>
      
    </table>
  </div>
  <!---->
</div>
`;
